### PR TITLE
support for cpu profiler and heap snapshot

### PR DESCRIPTION
**Pull request by rafixer**

### CODE REVIEW

**Comment by kwiatkk1** [comment]
Could you share any reasoning behind that feature?

On a first look, it seems that it may perfectly be a part of your own app and/or separate library -- no worker specific things here as far as I see. That's why describing motivation will help to understand what you're trying to achieve.

**Comment by rafixer** [comment]
> Could you share any reasoning behind that feature?
> 
> On a first look, it seems that it may perfectly be a part of your own app and/or separate library -- no worker specific things here as far as I see. That's why describing motivation will help to understand what you're trying to achieve.

Sure! I need to initialize CPU Profiler on worker that's actually has some task to do.

File: .travis.yml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 language: node_js
 node_js:
-  - "11.10"
+  - "12.18"
```
Comment (bgalek) [inline .travis.yml:3]: I've set LTS here, so this change is not needed anymore ;)

File: lib/util/cpu-profiler.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+const inspector = require('inspector');
+const fs = require('fs');
+
+const DEFAULT_DURATION = 30000; // 30 sec
```
Comment (bgalek) [inline lib/util/cpu-profiler.js:4]: maybe 30_SECONDS would be better name? (instead of comment)

File: lib/util/heap-profiler.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+class HeapProfiler {
+    constructor () {
+        this.session = new inspector.Session();
+        this.file = null;
+    }
+
+    async connect() {
+        this.session.connect();
+
+        this.file = await fs.open(`HeapSnapshot-${process.pid}-${Date.now()}.heapsnapshot`, 'w');
```
Comment (bgalek) [inline lib/util/heap-profiler.js:13]: where will this file be created? ./? or ./?lib/util/heapprofiler? or maybe wherever connect() is called?

File: lib/worker/child-loader.js [hunk 4]
```diff
@@ -9,12 +11,26 @@ function setupModule({ modulePath }) {
...
+    parentPort.on('message', callSplitter);
 
     // report readiness
     parentPort.postMessage('ready');
 }
 
+function callSplitter(requestData) {
+    const { cmd } = requestData;
+
+    if (cmd === 'profiler') {
```
Comment (bgalek) [inline lib/worker/child-loader.js:23]: any tests for these commands? :)

File: lib/util/cpu-profiler.js (hunk 2 above)
Comment (rafixer) [reply lib/util/cpu-profiler.js:4]: the name of variable is fine, unnecessery comment, so to delete ;)

File: lib/util/heap-profiler.js (hunk 3 above)
Comment (rafixer) [reply lib/util/heap-profiler.js:13]: on current working directory of Node.js process

Comment (rafixer) [reply lib/util/heap-profiler.js:13]: in our case, main application directory ;)

**Pull request merged**